app/controllers/admin/users_controller.rb

```diff
@@ -50,8 +66,14 @@ def edit; end
     private
 
     def user_params
-      params.require(:user).permit(:email, :name, :email_public, :biography, :nickname, :affiliation, :is_admin, :username, :login, :is_disabled,
+      # no password attribute in ICHAIN mode
+      if ENV['OSEM_ICHAIN_ENABLED'] == 'true'
+        params.require(:user).permit(:email, :name, :email_public, :biography, :nickname, :affiliation, :is_admin, :username, :login, :is_disabled,
```

Comment:
No need to do this all again. You can [merge](http://api.rubyonrails.org/classes/ActionController/Parameters.html#method-i-merge) `:user` into the params when `ENV['OSEM_ICHAIN_ENABLED']` is set.